src/promptflow/promptflow/_sdk/_service/entry.py

```diff
@@ -103,9 +103,9 @@ def validate_port(port, force_start):
     else:
         # Start a pfs process using detach mode
         if platform.system() == "Windows":
-            os.spawnv(os.P_DETACH, sys.executable, cmd)
+            subprocess.Popen(cmd, creationflags=subprocess.CREATE_NEW_PROCESS_GROUP)
         else:
-            os.system(" ".join(["nohup"] + cmd + ["&"]))
+            subprocess.Popen(cmd, start_new_session=True)
```

Comment:
add the reason we need to change this in PR description?  Pls also consider add comment here.

Reply:
investigating, I found pfs wil start failed in detach mode in the first time. not only msi, also python.

If I first call it using subprocess.call(cmd), when start it in detach mode, then it can start service successfully.

Reply:
![image](https://github.com/microsoft/promptflow/assets/26239730/757b6698-c65e-4936-93fc-df1df5b304d0)